pkg/antctl/command_definition.go

```diff
@@ -254,6 +255,93 @@ func (cd *commandDefinition) yamlOutput(obj interface{}, writer io.Writer) error
 	return nil
 }
 
+// respTransformer collects output fields in original transformedResponse
+// and flattens them. respTransformer realizes this by turning obj into
+// JSON and unmarshalling it.
+// E.g. agent's transformedVersionResponse will only have two fields after
+// transforming: agentVersion and antctlVersion.
+func respTransformer(obj interface{}) (interface{}, error) {
+	var jsonObj bytes.Buffer
+	err := json.NewEncoder(&jsonObj).Encode(obj)
+	if err != nil {
+		return nil, fmt.Errorf("error when encoding data in json: %w", err)
+	}
+	jsonStr := jsonObj.String()
+
+	var target interface{}
+	err = json.Unmarshal([]byte(jsonStr), &target)
+	if err != nil {
+		return nil, fmt.Errorf("error when unmarshalling data in json: %w", err)
+	}
+	return target, nil
+}
+
+func (cd *commandDefinition) tableOutput(obj interface{}, writer io.Writer) error {
+	target, err := respTransformer(obj)
+	if err != nil {
+		return fmt.Errorf("error when transforming obj: %w", err)
+	}
+
+	list, multiple := target.([]interface{})
+	var args []string
+	if multiple {
+		for _, el := range list {
+			m := el.(map[string]interface{})
+			for k := range m {
+				args = append(args, k)
+			}
+			break
+		}
+	} else {
+		m, _ := target.(map[string]interface{})
+		for k := range m {
+			args = append(args, k)
+		}
+	}
+
+	var buffer bytes.Buffer
+	for _, arg := range args {
+		buffer.WriteString(arg)
+		buffer.WriteString("\t")
+	}
+	attrLine := buffer.String()
+
+	var valLines []string
+	if multiple {
+		for _, el := range list {
+			m := el.(map[string]interface{})
+			buffer.Reset()
+			for _, k := range args {
+				buffer.WriteString(m[k].(string))
```

Comment:
You should not assert the field is always `string`, it will frequentely run to panic. As I questioned earlier,  how a struct value could be outputted by using this function? One option is to use `fmt.Sprint` to output struct.

Reply:
You are right. Updated the code.